modules/aggregation-layers/src/screen-grid-layer/screen-grid-cell-layer.js

```diff
@@ -0,0 +1,181 @@
+      instanceCounts: {size: 4, instanced: true, noAlloc: true}
+    });
+    this.setState({
+      model: this._getModel(gl)
+    });
+  }
+
+  updateState({oldProps, props, changeFlags}) {
+    super.updateState({oldProps, props, changeFlags});
+
+    if (oldProps.cellSize !== props.cellSize || changeFlags.viewportChanged) {
+      const attributeManager = this.getAttributeManager();
+      attributeManager.invalidateAll();
+    }
+
+    this._updateUniforms({oldProps, props, changeFlags});
+  }
+
+  draw({uniforms}) {
+    const {parameters = {}, maxTexture} = this.props;
+    const minColor = this.props.minColor || DEFAULT_MINCOLOR;
+    const maxColor = this.props.maxColor || DEFAULT_MAXCOLOR;
+
+    // If colorDomain not specified we use default domain [1, maxCount]
+    // maxCount value will be sampled form maxTexture in vertex shader.
+    const colorDomain = this.props.colorDomain || [1, 0];
+    const {model, cellScale, shouldUseMinMax, colorRange} = this.state;
+    const layerUniforms = {
+      minColor,
+      maxColor,
+      maxTexture,
+      cellScale,
+      colorRange,
+      colorDomain,
+      shouldUseMinMax
+    };
+
+    uniforms = Object.assign(layerUniforms, uniforms);
+    model.draw({
+      uniforms,
+      parameters: Object.assign(
+        {
+          depthTest: false,
+          depthMask: false
+        },
+        parameters
+      )
+    });
+  }
+
+  calculateInstancePositions(attribute, {numInstances}) {
+    const {width, height} = this.context.viewport;
+    const {cellSizePixels, numCol} = this.props;
+    const {value, size} = attribute;
+
+    for (let i = 0; i < numInstances; i++) {
+      const x = i % numCol;
+      const y = Math.floor(i / numCol);
+      value[i * size + 0] = ((x * cellSizePixels) / width) * 2 - 1;
+      value[i * size + 1] = 1 - ((y * cellSizePixels) / height) * 2;
+      value[i * size + 2] = 0;
+    }
+  }
+
+  // Private Methods
+
+  _getModel(gl) {
+    return new Model(
+      gl,
+      Object.assign({}, this.getShaders(), {
+        id: this.props.id,
+        geometry: new Geometry({
+          drawMode: GL.TRIANGLE_FAN,
+          attributes: {
+            positions: new Float32Array([0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0])
+          }
+        }),
+        isInstanced: true
+      })
+    );
+  }
+
+  _shouldUseMinMax() {
+    const {minColor, maxColor, colorDomain, colorRange} = this.props;
+    if (minColor || maxColor) {
+      log.deprecated('ScreenGridLayer props: minColor and maxColor', 'colorRange, colorDomain')();
+      return true;
+    }
+    // minColor and maxColor not supplied, check if colorRange or colorDomain supplied.
+    // NOTE: colorDomain and colorRange are experimental features, use them only when supplied.
+    if (colorDomain || colorRange) {
+      return false;
+    }
+    // None specified, use default minColor and maxColor
+    return true;
+  }
+
+  _updateUniforms({oldProps, props, changeFlags}) {
```

Comment:
- Avoid using named parameters in internal methods
- `changeFlags` not used

Reply:
Removed named params, `changeFlags` is used in this function.